elastic/kibana
# [Discover] Remove column from sorting array when removed from table

## Summary

When you removed a column in Discover, that was part of the sorting array, the sorting by this field was still active, and there was no way to remove it except modifying the URL. This caused confusion especially when you switched index pattern. The suggested solution is automatically removing the field from the sorting array, when you remove the column from the table.
Fixes #54345

### How to test

* Add a column to to the doc table, sort by the column. 
* Take a look at the URL, the field should be part of the `sort:` part.
* Remove the column.
* The URL should no longer contain the field in the `sort:` part.

### Checklist

Delete any items that are not applicable to this PR.
